src/lib.rs

```diff
@@ -821,8 +937,12 @@ impl Device {
 
     /// Creates a new buffer, maps it into host-visible memory, copies data from the given slice,
     /// and finally unmaps it, returning a [`Buffer`].
-    pub fn create_buffer_with_data(&self, data: &[u8], usage: BufferUsage) -> Buffer {
-        let mapped = self.create_buffer_mapped(data.len(), usage);
+    pub fn create_buffer_with_data(&self, data: &[u8], usage: BufferUsage, label: Option<String>) -> Buffer {
```

Comment:
I included `label` as an argument here instead of leaving it blank, but it does make it a bit less convenient to use

Reply:
Should we have `label` on the convenience methods? The descriptor could just be be used instead.

Reply:
I also think `create_buffer_mapped` will prefer CPU visible memory instead of device local - which should primarily be used for staging buffers. Not having the label is sort of a hint that this method isn't for your read-only long-lived data.

Reply:
Yeah we could omit it here but labels can also still be useful for staging buffers

Reply:
I think it's fine to have the argument on `create_buffer_mapped` but let's not put it on convenience methods (note to @aloucks  - in case of `create_buffer_with_data` one can't use a descriptor as a workaround, they need to call `create_buffer_mapped`).

In the nearest future hopefully the upstream API around mapping will be better, i.e. you'd just pass `mapped: true` in the descriptor, and then you'd obtain the mapping pointer through a different method.

> I also think create_buffer_mapped will prefer CPU visible memory instead of device local - which should primarily be used for staging buffers.

Filed https://github.com/gfx-rs/wgpu/issues/542 to make this better.